File: cash/cash.c
#include <cs50.h>
#include <stdio.h>
#include <math.h>

int main(void)
{
    int c=0;
    // c is for coin count

    float n;
    //n is for the chnage owed

    //prompting the user to enter the change owed
    do
    {
        n= get_float("hello there! please enter the change owed in dollars: ");

    }
    while(n<=0);

    //calculating the minimum number of changes
    //converting dollars to cents
    n = round(n*100);

    //number of quaters.
    while (n>=25)
    {

        n-=25;
        c++;
    }

    //number of dimes
    while (n>=10)
    {

        n-=10;
        c++;
    }

    //number of nickles
    while (n>=5)
    {

        n-=5;
        c++;
    }

    //number of pennies
    while (n>=1)
    {

        n-=1;
        c++;

    }
    printf("%i\n",c);

}

File: cash/cash.py
from cs50 import get_float
# prompt user for change owed
while True:
    change = get_float("Change owed: ")
    if change > 0.00:
        break

# convert it into cents
change *= 100
change = int(round(change, 0))


# keep track of no. of coins used
coins = 0

# use the largest coin possible
if change >= 25:
    coins += change//25
    change %= 25

if change >= 10:
    coins += change//10
    change %= 10

if change >= 5:
    coins += change//5
    change %= 5

if change >= 1:
    coins += change

# print final no.
print(coins)
